<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [ ]:
# Membuat dataframes dari dictionary
import pandas as pd
import numpy as np
 
nama_umur = {'Nama' : ['Agung', 'Bang jago', 'Dodi', 'Hanif', 'Ikhsan'],
'Umur' : [32, 55, 20, 43, 30]}
data_frame = pd.DataFrame(nama_umur)
print (data_frame)

        Nama  Umur
0      Agung    32
1  Bang jago    55
2       Dodi    20
3      Hanif    43
4     Ikhsan    30


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [ ]:
# Membaca data dan memasukannya kedalam suatu variabel
import pandas as pd

data = pd.read_csv(r'diamonds-1.csv')

print(data)

       Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]


Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 
1. Pada Series, index pada data bisa kita ubah urutannya dan bisa diubah dengan string, sementara list tidak bisa diubah indexnya. Index dictionary dapat menggunakan kata kunci, bukan dengan urutan angka ataupun huruf.
2. Series adalah objek yang hanya terdiri dari satu dimensi dan mampu menyimpan berbagai jenis file seperti integer, string, dan sebagainya. Sementara dataframne terdiri dari dua dimensi yang terdiri dari index baris dan index kolom.

In [ ]:
# code here
import pandas as pd

a = pd.Series(['a', 'b', 'c', 'd'], index = ['satu', 'dua', 'tiga', 'empat'])
print (a)

satu     a
dua      b
tiga     c
empat    d
dtype: object


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [ ]:
# code here
import pandas as pd
harga = data['price'].head(1)
warna = data['color'].tail(1)

print(harga)
print(warna)


0    326
Name: price, dtype: int64
53939    D
Name: color, dtype: object


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: Statistika deskriptif adalah suatu metode penyajian data untuk mendapatkan informasi yang kita inginkan

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [ ]:
# code here
import pandas as pd
rata = pd.DataFrame(data, columns=['price']).mean()
print(rata)
standar = pd.DataFrame(data, columns=['depth']).std()
print(standar)
maks = pd.DataFrame(data, columns=['carat']).max()
print(maks)


price    3932.799722
dtype: float64
depth    1.432621
dtype: float64
carat    5.01
dtype: float64


Jawab:
Rata-rata price = 3932.799722
Standar deviasi depth = 1.432621
Nilai max carat = 5.01